libraries/chain/include/eosio/chain/chain_config.hpp

```diff
@@ -184,6 +184,60 @@ struct chain_config_v1 : chain_config_v0 {
    }
 };
 
+/**
+ * @brief v2 Producer-voted blockchain configuration parameters
+ *
+ * If Adding new parameters create chain_config_v[n] class instead of adding
+ * new parameters to v2, v1 or v0. This is needed for snapshots backward compatibility
+ */
+struct chain_config_v2 : chain_config_v1 {
+   using Base = chain_config_v1;
+
+   uint32_t   max_sync_call_depth = config::default_max_sync_call_depth;            ///< size limit for sync call depth
+   uint32_t   max_sync_call_data_size = config::default_max_sync_call_data_size;    ///< size limit for sync call data: input and return value respectively
+
+   //order must match parameters as ids are used in serialization
+   enum {
+     max_sync_call_depth_id = Base::PARAMS_COUNT,
+     max_sync_call_data_size_id,
+     PARAMS_COUNT
+   };
+
+   inline const Base& base() const {
+      return static_cast<const Base&>(*this);
+   }
+
+   void validate() const;
+
+   template<typename Stream>
+   friend Stream& operator << ( Stream& out, const chain_config_v2& c ) {
+      return c.log(out) << "\n";
+   }
+
+   friend inline bool operator == ( const chain_config_v2& lhs, const chain_config_v2& rhs ) {
+      //add v1 parameters comarison here
+      return ( std::tie(lhs.max_sync_call_depth, lhs.max_sync_call_data_size) ==
+               std::tie(lhs.max_sync_call_depth, rhs.max_sync_call_data_size) ) &&
+             lhs.base() == rhs.base();
+   }
+
+   friend inline bool operator != ( const chain_config_v2& lhs, const chain_config_v2& rhs ) {
+      return !(lhs == rhs);
+   }
```

Comment:
`operator!=` should not be needed with c++20.

Reply:
> `operator!=` should not be needed with c++20.

No need to define `operator ==()` when `= default;` is the same thing.

Reply:
Thanks. Removed. 

https://github.com/AntelopeIO/spring/pull/1257/commits/93f0295e3ab8e056ab42270d2a28620bcee562e4

Reply:
Don't we need the `= default` still?  I think this will skip equality check of v2 members as is